File: src/main/java/org/books/services/Report.java
package org.books.services;

public enum Report {

    WORKS("works.jrxml"),
    BOXES("boxes.jrxml"),
    LIBRARY("library.jrxml");

    private static final String CLASSPATH = "classpath:reports/";

    private final String file;

    Report(String file) {
        this.file = file;
    }

    public String getResourceName() {
        return String.format("%s%s", CLASSPATH, file);
    }
}

File: src/main/java/org/books/services/Report.kt
package org.books.services

enum class Report private constructor(private val file: String) {

    WORKS("works.jrxml"),
    BOXES("boxes.jrxml"),
    LIBRARY("library.jrxml");

    val resourceName: String
        get() = String.format("%s%s", CLASSPATH, file)

    companion object {

        private val CLASSPATH = "classpath:reports/"
    }
}
